frontend/src/components/DocumentDetails/DocumentDetails.tsx

```diff
@@ -0,0 +1,63 @@
+import React from "react";
+import { formatDistanceToNow, parseISO } from "date-fns";
+import { Address, Status, Part } from "solo-types";
+
+interface DocumentDetailDataProps {
+  statuses: Status[];
+  part?: Part;
+  shipper?: Address;
+  receiver?: Address;
+}
+
+const DocumentDetails: React.FC<DocumentDetailDataProps> = ({
+  statuses,
+  part,
+  shipper,
+  receiver
+}) => {
+  return (
+    <div className="grid-row flex-justify flex-align-start padding-5">
+      <div className="flex-col">
+        <div className="text-bold">DIC</div>
+        {statuses.map(({ status_date, dic: { code } }) => {
+          const formattedDate = formatDistanceToNow(parseISO(status_date));
+          return (
+            <div
+              className="grid-row flex-nowrap flex-col flex-align-center"
+              key={code}
+            >
+              <div className="margin-right-2">{code}</div>
+              <div>{`${formattedDate} ago`}</div>
+            </div>
+          );
+        })}
+      </div>
+      {part && (
+        <>
+          <div className="flex-col">
+            <div className="text-bold">NIIN</div>
+            <div>{part.nsn}</div>
+          </div>
+          <div className="flex-col">
+            <div className="text-bold">Unit of Measure</div>
+            <div>{part.uom}</div>
+          </div>
+        </>
+      )}
+      {shipper && (
+        <div className="flex-col">
+          <div className="text-bold">Shipped From</div>
+          <div>{shipper.name}</div>
+        </div>
+      )}
+      {receiver && (
+        <div className="flex-col">
+          <div className="text-bold">Shipped To</div>
+          <div>{receiver.name}</div>
+        </div>
+      )}
+    </div>
```

Comment:
This is a UI/UX question. Should we display the spots with empty data to imply there isn't data for the user or should we hide them. I haven't thought much about this but would love your feedback.

Reply:
What I mean by that is should the title be there no matter what.

Reply:
Agreed, updated.